### DISCUSSION TITLE
How to interpolate markdown into markdown?

### DISCUSSION THREAD

**Discussion by Kurohyou** [vuejs/vitepress#2921] [Q&A] [answered]
For the doc site I'm building, I have some data that is being loaded using [build time data loading](https://vitepress.dev/guide/data-loading#build-time-data-loading). One of the properties on the data that is loaded is a description field that can potentially have markdown (`[text](url)`) or JSDOC (`{@link elementName}`) style links in it. Some pseudo description content:

```
Function that creates a [demo link](https://my-link.com). An alias for {@link otherFunc}.
```

How could I convert those to actual `<a>` tags to provide actual links to other pieces of the site (in the case of the JSDOC link syntax, or to external pages (in the case of the markdown syntax links)?

What I have tried so far:
```md
## {{ obj.name }}
{{obj.kind !== 'Mixin' ? obj.kind : ''}}

{{ obj.description }}
```
Which obviously doesn't work.

Edit: and actually this text blob might really have any basic markdown syntax contained within it.

**Comment by Kurohyou** [comment]
anyone have a suggestion on how to handle this?

**Comment by brc-dd** [comment, marked as answer]
Try something like this from your loader file:

```ts
import { createMarkdownRenderer } from 'vitepress'

const config = globalThis.VITEPRESS_CONFIG
const md = await createMarkdownRenderer(config.srcDir, config.markdown, config.site.base, config.logger)

// return html from here -- do md.render(foo)
```

Now in your markdown file, you can inject that html: 

```md
<script setup>
import { foo } from 'your/foo.data.js'
</script>

<div v-html="foo" />
```

**Reply by Kurohyou** [reply]
That looks like it would work, but vitepress doesn't appear to have an export for `createMarkdownRenderer`?

**Reply by brc-dd** [reply]
> vitepress doesn't appear to have an export for createMarkdownRenderer?

It has in node API. Won't work in your client code. You'll need to keep that in your loader file.

**Reply by Kurohyou** [reply]
ah!

**Reply by Kurohyou** [reply]
Ok, this works great! Thanks brc-dd
